### ISSUE TITLE
Batch Processing - Issues

### ISSUE THREAD

**Issue by MrKuenning** [dinilvalsan/random-pose-generator#3]
I love the idea of batch-processing my many collections of openpose images.

I am however running into some issues.
I filled in the details and pointed it at a folder with 30 pose images in it.

It spat out 30 beautiful images in a generated folder, but they did not correlate to the poses at all.
Might as well have been random poses, because not one of the 30 images lined up with the pose.

Also, it destroyed the original 30 pose images.  It moved them one at a time into the generated folder but the moved image was sized at zero bytes and was no longer a viewable image.

I do want to figure out how to get it to work.

My setup is local processing but published to an external URL. (Example: https://ai.domain.com) It clearly communicated with the webUI, as it could see my sampler options and it did spit out images following the prompt.
I do have Openpose installed as an extension.
I am running the current builds of both.

**Comment by dinilvalsan** [comment]
@MrKuenning are you sure controlnet extension is installed in the instance with openpose model there?
Without tool can you make sure you can generate image with openpose with automatic 1111 as expected? Only after that plugin this tool... just my guess, not sure about extend of the problem your facing

**Comment by dinilvalsan** [comment]
@MrKuenning One more guess  can you please try with these poses?

![117](https://user-images.githubusercontent.com/38812233/230755715-3e35527b-640d-493f-a3b9-11d14b6b5ba9.jpeg)
![132](https://user-images.githubusercontent.com/38812233/230755718-ecac9d79-d379-46bf-8800-13a9e2487776.jpeg)
![133](https://user-images.githubusercontent.com/38812233/230755719-f72f7837-8474-4ed5-888c-e208f16689f4.jpeg)
![216](https://user-images.githubusercontent.com/38812233/230755720-fc140c55-2a2a-40d4-b65c-8c8f6bee6e0a.jpeg)

**Comment by dinilvalsan** [comment]
@MrKuenning if above images work, i am guessing it might be file extension issues, as you can see  
[ image_extensions = [".png", ".jpg", ".jpeg"]](https://github.com/dinilvalsan/random-pose-generator/blob/4b47391bd072e4c95cd860c72aa937357e1635fe/batch-processing.py#L35) 

If that doesn't fix, please attach screenshot of console error if possible

**Comment by MrKuenning** [comment]
So here is the output folder with the broken pose files and the 4 generated poses.

![2023-04-08 21_39_39-generated](https://user-images.githubusercontent.com/16549268/230756430-74045b80-9ab1-45df-bb7c-96878aac4ebf.jpg)

For contrast, here is what I get running it manually in controlnet.

![2023-04-08 21_44_59-generated](https://user-images.githubusercontent.com/16549268/230756590-04a9eb04-404b-42db-9838-f5af26b02b0c.jpg)

Here is the console output:

```
['Canny Difference [ea6e3b9c]', 'Canny [e3fe7712]', 'Depth Difference [978ef0a1]', 'Depth [400750f6]', 'Hed Difference [86db5d7c]', 'Hed [13fee50b]', 'Mlsd Difference [14f0845a]', 'Mlsd [e3705cfa]', 'Norma Ldifference [00173cc1]', 'Normal [63f96f7c]', 'Openpose Difference [1723948e]', 'Openpose [9ca67cc5]', 'Scribble Difference [1f29174d]', 'Scribble [c508311e]', 'Seg Difference [a1e85e27]', 'Seg [b9c1cc12]', 'T2IAdapter_t2iAdapter BodyPose [1db80368]', 'T2IAdapter_t2iAdapter Canny [ce056dd0]', 'T2IAdapter_t2iAdapter Color [c58d39ff]', 'T2IAdapter_t2iAdapter Depth [d0e98e8d]', 'T2IAdapter_t2iAdapter Keypose [d9e84085]', 'T2IAdapter_t2iAdapter Openpose [80482077]', 'T2IAdapter_t2iAdapter Seg [6ccf07d4]', 'T2IAdapter_t2iAdapter Sketch [e33afc93]', 'T2IAdapter_t2iAdapter Style [892c9244]', 't2iadapter_canny_sd14v1 [80bfd79b]', 't2iadapter_color_sd14v1 [8522029d]', 't2iadapter_depth_sd14v1 [fa476002]', 't2iadapter_keypose_sd14v1 [ba1d909a]', 't2iadapter_openpose_sd14v1 [7e267e5e]', 't2iadapter_seg_sd14v1 [6387afb5]', 't2iadapter_sketch_sd14v1 [e5d4b846]', 't2iadapter_style_sd14v1 [202e85cc]']
Loading model: t2iadapter_openpose_sd14v1 [7e267e5e]
Loaded state_dict from [E:\AI\A1111\models\ControlNet\t2iadapter_openpose_sd14v1.pth]
Error running process: E:\AI\A1111\extensions\sd-webui-controlnet\scripts\controlnet.py
Traceback (most recent call last):
  File "E:\AI\A1111\modules\scripts.py", line 417, in process
    script.process(p, *script_args)
  File "E:\AI\A1111\extensions\sd-webui-controlnet\scripts\controlnet.py", line 679, in process
    model_net = self.load_control_model(p, unet, unit.model, unit.low_vram)
  File "E:\AI\A1111\extensions\sd-webui-controlnet\scripts\controlnet.py", line 468, in load_control_model
    model_net = self.build_control_model(p, unet, model, lowvram)
  File "E:\AI\A1111\extensions\sd-webui-controlnet\scripts\controlnet.py", line 510, in build_control_model
    network = network_module(
  File "E:\AI\A1111\extensions\sd-webui-controlnet\scripts\adapter.py", line 82, in __init__
    self.control_model.load_state_dict(state_dict)
  File "E:\AI\A1111\venv\lib\site-packages\torch\nn\modules\module.py", line 1671, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for Adapter:
        size mismatch for conv_in.weight: copying a param with shape torch.Size([320, 192, 3, 3]) from checkpoint, the shape in current model is torch.Size([320, 64, 3, 3]).

100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 15.00it/s]
```

**Comment by MrKuenning** [comment]
Well I may have figured out part of it.  It says its loading `t2iadapter_openpose_sd14v1.pth`  that model has never worked for me.  I use eigther `Openpose.safetensors` or `Openpose Difference.safetensors`.

My above manual run was using OpenPose Difference.

**Comment by dinilvalsan** [comment]
yes got the issue, 

see in the code

for model in model_list['model_list']:
                if 'openpose' in model:
                    model_name = model
                    break
                    
  t2iadapter_openpose gets executed

**Comment by dinilvalsan** [comment]
so can you try uninstalling t2iadapter? if not i may need to update to handle this

**Comment by dinilvalsan** [comment]
@MrKuenning please give me some time to reflect on proper solution

**Comment by MrKuenning** [comment]
I removed the model t2iadapter_openpose_sd14v1.pth from my controlnet model folder and closed and opened both the webUI as well as the batch processor.

Running again with one preview file, I got the message "Opempose Controlnet not detected"

And this in the console:

```
['Canny Difference [ea6e3b9c]', 'Canny [e3fe7712]', 'Depth Difference [978ef0a1]', 'Depth [400750f6]', 'Hed Difference [86db5d7c]', 'Hed [13fee50b]', 'Mlsd Difference [14f0845a]', 'Mlsd [e3705cfa]', 'Norma Ldifference [00173cc1]', 'Normal [63f96f7c]', 'Openpose Difference [1723948e]', 'Openpose [9ca67cc5]', 'Scribble Difference [1f29174d]', 'Scribble [c508311e]', 'Seg Difference [a1e85e27]', 'Seg [b9c1cc12]', 'T2IAdapter_t2iAdapter BodyPose [1db80368]', 'T2IAdapter_t2iAdapter Canny [ce056dd0]', 'T2IAdapter_t2iAdapter Color [c58d39ff]', 'T2IAdapter_t2iAdapter Depth [d0e98e8d]', 'T2IAdapter_t2iAdapter Keypose [d9e84085]', 'T2IAdapter_t2iAdapter Openpose [80482077]', 'T2IAdapter_t2iAdapter Seg [6ccf07d4]', 'T2IAdapter_t2iAdapter Sketch [e33afc93]', 'T2IAdapter_t2iAdapter Style [892c9244]', 't2iadapter_canny_sd14v1 [80bfd79b]', 't2iadapter_color_sd14v1 [8522029d]', 't2iadapter_depth_sd14v1 [fa476002]', 't2iadapter_keypose_sd14v1 [ba1d909a]', 't2iadapter_seg_sd14v1 [6387afb5]', 't2iadapter_sketch_sd14v1 [e5d4b846]', 't2iadapter_sketch_sd15v2 [f5789421]', 't2iadapter_style_sd14v1 [202e85cc]', 't2iadapter_zoedepth_sd15v1 [44291dde]']
```

Maybe it doesn't like that I renamed my models?

**Comment by dinilvalsan** [comment]
hold on 10mins

**Comment by MrKuenning** [comment]
It would be nice to have the option to choose which controlnet model I want to process with.
Also would be nice to be able to specify an output file format, as I typically generate jpg files for space reasons.

**Comment by dinilvalsan** [comment]
Can you please check with this build https://drive.google.com/file/d/1RwFRKA05EPTlaRZ2SfEwEUfWoY5JYiRh/view?usp=share_link

yes, in next update i will try it :)

**Comment by MrKuenning** [comment]
That build would not let me extract. Windows immediately said it had a virus and deleted it.
Tried multiple times.  Don't know if it was missing a signature or something.

**Comment by dinilvalsan** [comment]
Ok, do you know how to build using virtualenv and pyinstaller?

After cloning repo & setting up virtualenv, run this command in terminal
pyinstaller --onefile --windowed batch-processing.py

this will generate exe, 
for temporary solution, can you disable antivirus and check, sorry this not the way i know it, but i need to getting signed as you mentioned, i will look it to it when i get time.

the virus issue is due to this 
https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg

**Comment by MrKuenning** [comment]
Sorry, I wish I knew how to compile.  That is an area I have not branched out into.  Maybe someday.

I will be happy to test the next signed one and provide feedback.

**Comment by dinilvalsan** [comment]
@MrKuenning sure, i will you updated on this

**Comment by MrKuenning** [comment]
Just tested v2 and it now follows the poses.  I do, have some other ideas which I will put in a new thread.  Thanks!
